examples/p4est_3d_dgsem/elixir_euler_baroclinic_instability.jl

```diff
@@ -190,7 +239,35 @@ semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver,
 # ODE solvers, callbacks etc.
 
 tspan = (0.0, 10 * 24 * 60 * 60.0) # time in seconds for 10 days
-ode = semidiscretize(semi, tspan)
+
+# Save RHS of the steady state and subtract it in every RHS evaluation.
+# This trick preserves the steady state exactly (to machine rounding errors, of course).
+# Otherwise, this elixir produces entirely unusable results for a resolution of 8x8x4 cells
+# per cube face with a polydeg of 3.
+# With this trick, even the polydeg 3 simulation produces usable (although badly resolved) results,
+# and most of the grid imprinting in higher polydeg simulation is eliminated.
+#
+# See https://github.com/trixi-framework/Trixi.jl/issues/980 for more information.
+u_steady_state = compute_coefficients(steady_state_baroclinic_instability, 0.0, semi)
+# Use a `let` block for performance (otherwise du_steady_state will be a global variable)
+let du_steady_state = similar(u_steady_state)
+  # Save RHS of the steady state
+  Trixi.rhs!(du_steady_state, u_steady_state, semi, 0.0)
+
+  global function corrected_rhs!(du, u, semi, t)
+    # Normal RHS evaluation
+    Trixi.rhs!(du, u, semi, t)
+    # Correct by subtracting the steady-state RHS
+    Trixi.@trixi_timeit Trixi.timer() "rhs correction" begin
+      # Use Polyester.@batch (that's what we use in Trixi.@threaded) for threaded performance
+      @batch for i in eachindex(du)
```

Comment:
Is there any way to avoid `ArgumentError: Package Polyester not found in current path` by using the fact that Trixi imports `Polyester: @batch`? I tried using `Trixi.@threaded`, but that gives me
```
ERROR: LoadError: LoadError: UndefVarError: @batch not defined
in expression starting at /home/erik/git/Trixi.jl/src/auxiliary/auxiliary.jl:188
in expression starting at /home/erik/git/Trixi.jl/examples/p4est_3d_dgsem/elixir_euler_baroclinic_instability.jl:252
```

Reply:
Currently, `@threaded` is delicate because of https://github.com/JuliaLang/julia/issues/23221 and https://github.com/JuliaLang/julia/issues/37691. I would like to avoid making the implementation based on `@batch` part of our official API. I tried to implement a workaround that let's you use `@threaded` outside of Trixi.

Reply:
Thanks! I implemented your fix from #982 in this PR to speed things up.